### ISSUE TITLE
Tab completion stopped working in IPython 6.3.0

### ISSUE THREAD

**Issue by rayosborn** [ipython/ipython#11070]
I have a class with the ```__dir__``` function defined to enable autocompletion. It works well in IPython 6.2.1, but has stopped working in IPython 6.3.0. I tested it in a newly updated Jupyter QtConsole in Python 3.6.
```
Jupyter QtConsole 4.3.1
Python 3.6.3 | packaged by conda-forge | (default, Nov  4 2017, 10:13:32) 
Type 'copyright', 'credits' or 'license' for more information
IPython 6.3.0 -- An enhanced Interactive Python. Type '?' for help.
```
Do I have to update any other modules that are not listed as a dependency?

Edit: This is for tab-completion of class attributes. Tab-completion seems to work for dictionaries.

**Comment by takluyver** [comment]
It's an issue in Jedi, the library we're moving towards for using completions. It's been fixed, but it needs a new release:

https://github.com/davidhalter/jedi/issues/1027

In the meantime, you can configure `Completer.use_jedi = False` to switch back to completions found by IPython's own machinery.

**Comment by rayosborn** [comment]
Thanks for the quick response. Should I leave this issue open for now to alert others? You're welcome to close it if you prefer.

**Comment by takluyver** [comment]
May as well leave it open for now, because people will probably open more if they don't see it.

**Comment by takluyver** [comment]
I've released IPython 6.3.1 which switches the default back to our own completions, so this issue should go away.
